DOC-719: Update mentions_select usage with profile templates (#1780)
tinymce/tinymce-docs: plugins/premium/mentions.md

### `mentions_select`

> This option enables an element to be presented below a mention when a user hovers over it on the page. This could include details about the user. A custom hover element can be created or a predefined template can be specified.

Reviewer: "element" -> "popup"?

Also, is it always above, or can it appear below?

"on the page" -> "in the editor"? This implied it would work on the published content, which I believe people would have to set up themselves separately?

Reviewer: ```suggestion
This option enables a hover card to be presented when a user hovers over a mention in {{site.productname}}. This could include details about the user. A custom hover card can be defined or a predefined template can be specified.
```

Reviewer: We went looking and "hover card" seems to be a common enough term - e.g. Chrome and Facebook use it, and they're similar concepts.

Author: Hover card sounds better, thanks for the suggested changes.

      userRequest[id] = fetch('/user?id=' + id);
    }
    userRequest[id].then(function(userDetail) {

>       // Create a custom select element

Reviewer 2: Should this be `hover element`? As that's what it's referred to in the original sentence.

        '</div>'
      );
      success(div);
      // Or simply use the predefined 'profile' template.
      // success({ type: 'profile', user: userDetail });
    });
  }
});
```

#### `mentions_select` with predefined templates

{{site.requires_5_6v}}

> If `mentions_select` is resolved with an object specifying type and user details, a predefined select template will be used. The only valid template at this time is `'profile'`. For details on the user properties required for the `profile` template, see: [User properties](#userproperties).

Reviewer 2: Saying `at this time` is generally discouraged, as it can imply we are going to add additional types when we may not. It's also hard to tell when "at this time" was referring to.

I can't think of anything better, so tagging @tylerkelly13 as he knows all :stuck_out_tongue:

Reviewer 3: ```suggestion
If `mentions_select` is resolved with an object specifying the type and user details, a predefined select template will be used. To use the predefined template, set `mentions_select` to `'profile'`. For details on the user properties required for the `profile` template, see: [User properties](#userproperties).
```

Reviewer 3: My suggestion may or may not be better. we say "A predefined template", implying one or more. so stating how to use the only one should be good. not sure if that "the"  is correct, please correct if not.

Author: I reworded it to say "the" so as to not imply that there'll be more templates added in the future. `mentions_select` can no longer be set to `'profile'` so I refered to an example to clarify the usage.

        '</div>'
      );
      success(div);

>       // or simply use the predefined 'profile' template.

Reviewer 3: ```suggestion
      // or use the predefined 'profile' template.
```
Sorry, should have seen this earlier. "simply" is not required

{{site.requires_5_6v}}

> If `mentions_select` is resolved with an object specifying the type and user details, a predefined select template will be used. To use the predefined template, set `type` to `'profile'`. See [`Example: Using mentions_select`](#exampleusingmentions_select) for an example of how this is done. For details on the user properties required for the `profile` template, see: [User properties](#userproperties).

Reviewer 3: ```suggestion
If `mentions_select` is resolved with an object specifying the type and user details, a predefined select template will be used. To use the predefined template, set `type` to `'profile'`. For details on the user properties required for the `profile` template, see: [User properties](#userproperties).
```
That example doesn't really show how to set the option to a predefined template, it just mentions that you can, same as here.

Reviewer: Also, "predefined select template"? Should it just be "predefined template"?

Reviewer: Lee clarified what "select template" means, but it's not obvious to me. Would "hover template" or "popup template" be better?
